app/jobs/runtime/prune_completed_tasks.rb

```diff
@@ -13,10 +13,12 @@ def initialize(cutoff_age_in_days)
         def perform
           logger = Steno.logger('cc.background')
           logger.info('Cleaning up old TaskModel rows')
-
           tasks_to_delete = TaskModel.where(state: prunable_states).where(Sequel.lit('updated_at < ?', cutoff_age))
-          deleted_count   = Database::BatchDelete.new(tasks_to_delete).delete
+          task_labels_to_delete = TaskLabelModel.where(task: tasks_to_delete)
+          deleted_label_count = Database::BatchDelete.new(task_labels_to_delete).delete
+          deleted_count = Database::BatchDelete.new(tasks_to_delete).delete
```

Comment:
It might be good to wrap the deletions in a `transaction` block in case the final task deletion fails--then the label deletion could be rolled back.  

Here's an example of a transaction block with the Sequel ORM:

https://github.com/cloudfoundry/cloud_controller_ng/blob/ee25971edd577137ecc1e7104ad343e55043de9f/app/actions/buildpack_create.rb#L11-L23

(it doesn't matter which Model or resource you use to get to the `db` object)

Reply:
Added database transaction [in this commit.](https://github.com/cloudfoundry/cloud_controller_ng/pull/3612/commits/c817d92ab222193d5a917dd2341f93166a6ebead)